### PR TITLE
feat: advanced validation setup

### PR DESCRIPTION
**Pull request by dleard**
- [x] application_validation_function table
- [x] application_validation_result type
- [x] application_validation function (to iterate on all functions in validation_function table)
- [x] pgTap tests
- [x] summary page alert with notifications from each validation where `is_ok` is false
- [x] unit tests

### CODE REVIEW

File: schema/test/unit/computed_columns/application_validation_test.sql [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+with record as (select row(application.*)::ggircs_portal.application from ggircs_portal.application where id=1)
+      select * from ggircs_portal.application_validation((select * from record)) order by is_ok desc limit 1;
+
+select results_eq(
+  $$
+    with record as (select row(application.*)::ggircs_portal.application from ggircs_portal.application where id=1)
+      select count(*) from ggircs_portal.application_validation((select * from record))
+  $$,
+  $$
+    (select count(*) from ggircs_portal.application_validation_function)
```
Comment (pbastia) [inline schema/test/unit/computed_columns/application_validation_test.sql:54]: This could be hardcoded instead

**Review by pbastia — DISMISSED** [review]
Looks great!

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]
Looks great! Just a couple of changes to ensure that tests don't break when we start implementing validation functions

File: schema/test/unit/computed_columns/application_validation_test.sql [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+language 'plpgsql' stable;
+
+select test_helper.clean_ggircs_portal_schema();
+alter table ggircs_portal.ciip_user disable trigger _welcome_email;
+alter table ggircs_portal.application disable trigger _send_draft_application_email;
+select test_helper.create_test_users();
+select test_helper.mock_open_window();
+select test_helper.create_applications(1, True, True);
+
+insert into ggircs_portal.application_validation_function(validation_function_name, validation_description, validation_failed_message)
```
[the remark this answers is not in the document]
Comment (dleard) [reply schema/test/unit/computed_columns/application_validation_test.sql:39]: `select test_helper.clean_ggircs_portal_schema()` truncates all tables in the schema.

[the remark this answers is not in the document]
Comment (matthieu-foucault) [reply schema/test/unit/computed_columns/application_validation_test.sql:39]: indeed, didn't notice that!

**Review by matthieu-foucault — DISMISSED** [review]

**Comment by matthieu-foucault** [comment]
@dleard just needs a rebase then

**Review by matthieu-foucault — APPROVED** [review]

**Pull request merged**